core/src/test/scala/org/apache/spark/util/UtilsSuite.scala

```diff
@@ -35,7 +36,64 @@ import org.apache.hadoop.fs.Path
 import org.apache.spark.SparkConf
 
 class UtilsSuite extends FunSuite with ResetSystemProperties {
+  
+  test("timeConversion") {
+    // Test -1
+    assert(Utils.timeStringAsSec("-1") === -1)
```

Comment:
Negative values are used in one of the other test suites, specifically the ```StreamingContextSuite``` with ```spark.cleaner.ttl``` so we need to handle this case.